ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/location/CaptureLocationTaskViewModel.kt

```diff
@@ -17,21 +17,46 @@ package com.google.android.ground.ui.datacollection.tasks.location
 
 import android.content.res.Resources
 import android.location.Location
+import androidx.lifecycle.viewModelScope
 import com.google.android.ground.ui.datacollection.tasks.AbstractTaskViewModel
 import com.google.android.ground.ui.datacollection.tasks.location.CaptureLocationTaskResult.Companion.toCaptureLocationResult
 import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+/** Location lock states relevant for attempting to enable it or not. */
+enum class LocationLockState {
+  /** The default, unknown state. */
+  UNKNOWN,
+
+  /** The location lock was already enabled, or an attempt was made. */
+  ALREADY_ENABLED,
+
+  /** The location lock was not already enabled. */
+  NEEDS_ENABLE,
+
+  /** Trigger to enable the location lock. */
+  ENABLE,
+}
```

Comment:
Can we move this to a shared class as location lock is present in other fragments as well?

Reply:
I realized that because I was able to completely localize usage of this enum to one file, I marked it private instead. I'd prefer it be used for this specialized purpose rather than generally across the code base, if that's ok with you!